### PR TITLE
A4A: initail onboarding Modal

### PR DESCRIPTION
**Pull request by andrii-lysenko**
Related to https://github.com/Automattic/jetpack-genesis/issues/355

## Proposed Changes

In order to easier approach this task and not to grow it too big, this presents only basic modal, as well as some refactoring to improve the scalability

## Testing Instructions

- Navigate to any `/marketplace` page and check `i` button near Referral toggle.
- Clicking it should open a modal.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/components/guide-modal/style.scss
Comment (yashwin) [inline client/a8c-for-agencies/components/guide-modal/style.scss:1]: A few style suggestions:

- Can we avoid using `--studio-` variables?
- Can we use RTL-specific attribute names (margin-block-end)? 
- I think it is better to move the color codes as variable on top of the file if we don't have an existing one and use it below. WDYT?

**Review by yashwin — COMMENTED** [review]
I found a few alignment issues that I think we plan to fix in another PR. For example The buttons should be on the right.

There is also a scroll that appears on the modal:

Screenshot 2024-05-30 at 3 01 51 PM

**Comment by andrii-lysenko** [comment]
> I found a few alignment issues that I think we plan to fix in another PR. For example The buttons should be on the right.


Thank you @yashwin. Yes, these issues are planned to be fixed in the following PR. I just wanted to merge refactoring part quicker to avoid conflicts, so I would be able to work only inside that component

File: client/a8c-for-agencies/components/guide-modal/style.scss
Comment (andrii-lysenko) [reply client/a8c-for-agencies/components/guide-modal/style.scss:1]: >I think it is better to move the color codes as variable on top of the file if we don't have an existing one and use it below. WDYT?

Which color codes do you have in mind?

Comment (andrii-lysenko) [reply client/a8c-for-agencies/components/guide-modal/style.scss:1]: Oh, I got it, it was rough CSS I forgot to delete

Comment (andrii-lysenko) [reply client/a8c-for-agencies/components/guide-modal/style.scss:1]: Now it should be better. Added style adjustments.

**Review by yashwin — APPROVED** [review]
Thanks for addressing my comments. LGTM! 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/13833285

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @andrii-lysenko, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Manage your clients' products without the burden of managing the billing. Assemble a cart of products, send a request for payment to your clients, and make commissions based on what you sell.`
- `Ensure "Refer products" is toggled on, and add any mix of products to your cart.`
- `During checkout, add your clients' email address and a note about the invoice for the selected products.`
- `Once sent, your client will get the invoice delivered to their inbox. After they pay, you'll be able to assign the products to their site.`
- `Clients will be billed at the end of every month for their products. When they pay, you'll make commissions on those products, which you'll be able to manage under the Referrals section, soon.`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/13833285).

**Pull request merged**